lib/common/client/middleware/response/facility_parser.rb

```diff
@@ -33,30 +33,43 @@ def attribute_mappings(entry, mapping)
             attrs = entry['attributes']
             mapped_attrs = { 'address' => {}, 'services' => {} }
             mapping.slice('unique_id', 'name', 'classification', 'website').each do |key, value|
-              mapped_attrs[key] = attrs[value]
+              mapped_attrs[key] = clean(attrs[value])
             end
             %w[hours access feedback phone].each do |name|
               mapped_attrs[name] = nested(mapping[name], attrs)
             end
             mapped_attrs['address']['physical'] = nested(mapping['physical'], attrs)
             mapped_attrs['address']['mailing'] = nested(mapping['mailing'], attrs)
-            mapped_attrs['services']['benefits'] = nested(mapping['benefits'], attrs)
-            mapped_attrs['services']['last_updated'] = services_date(attrs)
-            mapped_attrs['services']['health'] = services_from_gis(mapping['services'], attrs)
-
-            mapped_attrs.merge!(
-              'lat' => entry['geometry']['y'],
-              'long' => entry['geometry']['x']
-            )
+            if mapping['benefits']
+              mapped_attrs['services']['benefits'] = {
+                'standard' => clean_benefits(nested(mapping['benefits'], attrs)),
+                'other' => attrs['Other_Services']
+              }
+
+            end
+            mapped_attrs['services']['last_updated'] = services_date(attrs) if mapping['services']
+            mapped_attrs['services']['health'] = services_from_gis(mapping['services'], attrs) if mapping['services']
+
+            mapped_attrs.merge!('lat' => entry['geometry']['y'], 'long' => entry['geometry']['x'])
           end
 
           def nested(item, attrs)
-            return unless item
+            return {} unless item
             item.each_with_object({}) do |(key, value), hash|
-              hash[key] = value.respond_to?(:call) ? value.call(attrs) : attrs[value]
+              hash[key] = value.respond_to?(:call) ? value.call(attrs) : clean(attrs[value])
+            end
+          end
+
+          def clean_benefits(benefits_hash)
+            benefits_hash.each_with_object([]) do |(key, value), list|
+              list << key if value == 'YES'
             end
           end
 
+          def clean(value)
```

Comment:
Is the idea of calling this method clean that we might need to do something more than stripping whitespace now or in the future? I'm not at all opposed to the method name if that's the case. Perhaps if you find yourself having many of these it might be useful to create a Util module with "class methods." If you did that, I suspect you would want a more specific name than "clean."

Reply:
good point, clean is very generic but I don't think we'll be doing other operations on it.. would it be confusing to just call it strip?

Reply:
i think a custom strip makes sense to me.